### PR TITLE
Updated director toxic classifier task

### PR DESCRIPTION
**Pull request by leox1v**
Updated toxic classifier task for the director model:
* now supports cpu-only evaluation (needed during multi-gpu training)
* added additional flags to support truncation and limited number of examples
* smaller fixes and adjustments

### CODE REVIEW

File: projects/director/tasks/safety.py [hunk 1]
```diff
@@ -182,32 +197,59 @@ def add_cmdline_args(
...
+                    opt['eval_classifier_model_file'],
+                    opt_overrides={
+                        'datatype': 'valid',
+                        'no_cuda': not opt['eval_classifier_use_cuda'],
+                    },
                 )
+                self.classifier.opt.log()
             else:
-                self.classifier = shared['classifier']
+                print('Load the classifier from shared')
```
Comment (klshuster) [inline projects/director/tasks/safety.py:241]: nit: can we use logging here?

```python
import parlai.utils.logging as logging
logging.info(...)
```

File: projects/director/tasks/safety.py [hunk 2]
```diff
@@ -182,32 +197,59 @@ def add_cmdline_args(
...
             DEFAULT_DELIM = '\n'
             self.delimiter = opt.get('delimiter', DEFAULT_DELIM)
         else:
             self.classifier = None
-        super().__init__(opt, shared)
 
     def share(self):
         shared = super().share()
-        shared['classifier'] = self.classifier
+        if self.classifier:
```
Comment (klshuster) [inline projects/director/tasks/safety.py:251]: are there instances where this is not instantiated?

**Review by klshuster — APPROVED** [review]

**Comment by klshuster** [comment]
could you also rebase on main to see if your teacher tests pass?

File: projects/director/tasks/safety.py (hunk 1 above)
Comment (leox1v) [reply projects/director/tasks/safety.py]: Done.

File: projects/director/tasks/safety.py (hunk 2 above)
Comment (leox1v) [reply projects/director/tasks/safety.py]: No. I removed it and added an assertion.

**Comment by leox1v** [comment]
> could you also rebase on main to see if your teacher tests pass?

I did. But now somehow all the intermediate commits show up here. Is there a way to get rid of this?

**Review by klshuster — APPROVED** [review]
thanks, failing tests are unrelated

**Pull request merged**